rules/library.bzl

```diff
@@ -991,31 +991,32 @@ def apple_library(name, library_tools = {}, export_private_headers = True, names
         )
         lib_names.append(swift_libname)
 
-        # Add generated swift header to header maps for angle bracket imports
-        swift_doublequote_hmap_name = name + "_swift_doublequote_hmap"
-        headermap(
-            name = swift_doublequote_hmap_name,
-            namespace = namespace,
-            hdrs = [],
-            direct_hdr_providers = [swift_libname],
-            tags = _MANUAL,
-            testonly = testonly,
-        )
-        private_dep_names.append(swift_doublequote_hmap_name)
-        _append_headermap_copts(swift_doublequote_hmap_name, "-iquote", additional_objc_copts, additional_swift_copts, additional_cc_copts)
-
-        # Add generated swift header to header maps for double quote imports
-        swift_angle_bracket_hmap_name = name + "_swift_angle_bracket_hmap"
-        headermap(
-            name = swift_angle_bracket_hmap_name,
-            namespace = namespace,
-            hdrs = [],
-            direct_hdr_providers = [swift_libname],
-            tags = _MANUAL,
-            testonly = testonly,
-        )
-        private_dep_names.append(swift_angle_bracket_hmap_name)
-        _append_headermap_copts(swift_angle_bracket_hmap_name, "-I", additional_objc_copts, additional_swift_copts, additional_cc_copts)
+        if objc_sources or cpp_sources:
```

Comment:
Is this change necessary / related to the indexing bit? Does it just dump garbage targets?

Reply:
I hit an issue in a pure swift app where the indexing `swifc` invocation would ask for this hmaps but they were never created causing the indexing invocation to fail. Will add a test case for this.

Reply:
I can't seem to repro this 🤔, I was in the middle of debugging something when I saw so maybe it was bad local state or something else that I missed.

Will revert for now, if we hit other issues in the future I can follow up and propose a proper fix.

Reply:
I think CI's repro-ed what I was talking about: https://github.com/bazel-ios/rules_ios/actions/runs/4307287081/jobs/7512162662
```sh
Make sure hmap files exist after build
File at ../../../bazel-out/ios-x86_64-min10.0-applebin_ios-ios_x86_64-dbg-ST-d31cf6d5fbab/bin/tests/ios/frameworks/dynamic/b/b_swift_doublequote_hmap.hmap not found!
```

That dynamic fmw is not going to produce `_swift_doublequote_hmap` since the `objc_library` is never built due to this target not having ObjC sources. 

I might have to bring back the conditional I had added here. Will double check if the test is searching for a valid list of hmaps first.

Reply:
That said, I'm bringing back the conditional.